src/app/pages/poll-participants/poll-participants.component.ts

```diff
@@ -52,13 +52,33 @@ export class PollParticipantsComponent implements OnInit, OnDestroy {
       // Connect to WebSocket
       const subscription = this.websocketService.establishConnection(this.slug);
       subscription.subscribe(pollItem => {
-        // Load poll
-        this.pollService.get(pollItem.pollId).subscribe(poll => this.poll = poll);
-        // Update UI
-        this.activeItemType = pollItem.type;
-        delete pollItem.type;
-        this.activeItem = pollItem;
+        if (Object.keys(pollItem).length > 1) {
+          // Load poll
+          if (!this.poll) {
+            this.pollService.get(pollItem.pollId).subscribe(poll => this.poll = poll);
+          } else {
+            this.poll.currentItem = pollItem.itemId;
+          }
+          // Randomize selection options if it is a quiz item
+          if (pollItem.type === 'quiz') {
```

Comment:
Outsource this to an enum for the future. This makes it easier if there are changes in the API.

Reply:
No need to do this right now, maybe just keep this in mind...

Reply:
Jep. Let's do this later on ...